plugins/eventMacro/eventMacro/Runner.pm

```diff
@@ -1407,13 +1407,12 @@ sub parse_call {
 
 	my $macro_name   = $call_command;
 	my $repeat_times = 1;
-	if ( $call_command =~ /\s/ ) {
+	if ( $call_command =~ /\s+/ ) {
 	    my @params;
 		( $macro_name, @params ) = parseArgs( $call_command );
 
 		# Update $.paramN with the values from the call.
 		$eventMacro->set_scalar_var( ".param$_", $params[ $_ - 1 ], 0 ) foreach 1 .. @params;
-		$eventMacro->set_scalar_var( ".param$_", undef,             0 ) foreach ( @params + 1 ) .. 100;
```

Comment:
This line should not be removed. It undefines params which weren't passed. If we remove this, then the following will do unexpected things.

```
macro foo {
  call bar 1 2
  call bar a
}
macro bar {
  log 1: $.param1
  log 2: $.param2
}
```

After this change, the code above will unexpectedly log `2` on the second call to `bar`:
```
1: 1
2: 2
1: a
2: 2
```

Reply:
oh, you also right about that, i'll return this line of code to there.